### ISSUE TITLE
[@types/ember bug] - set() is not the same as this.set()

### ISSUE THREAD

**Issue by nightire** [typed-ember/ember-cli-typescript#308]
### Which package(s) does this problem pertain to?
  - [x] @types/ember

### What are instructions we can follow to reproduce the issue?
``` typescript
import EmberObject, { set } from '@ember/object';

export class Foo extends EmberObject {
  name!: string;

  changeName(name: string) {
    set(this, 'name', name);
  }
}
```
### Now about that bug. What did you expect to see?
It is expected to has no errors.

### What happened instead?
Typescript throws an error:

```
Argument of type 'string' is not assignable to parameter of type UnwrapComputedPropertySetter<this["name"]>'.
```

#### Note
1. use `name: any` instead of `name: string`, error will be gone
2. use `this.set` instead of `set()`, error will be gone

**Comment by mike-north** [comment]
Fix is ready for merge here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/28915

New `@types/ember@~3.0` version should be cut early next week

**Comment by mike-north** [comment]
Released
